webapp/src/components/modals/channel_settings/select_channel_subscription.tsx

```diff
@@ -24,22 +24,24 @@ export default class SelectChannelSubscriptionInternal extends React.PureCompone
         subscriptionToDelete: null,
     };
 
-    handleDeactivateCancel = () => {
+    handleCancelDelete = () => {
         this.setState({showConfirmModal: false});
     }
 
-    handleConfirmDelete = (sub: ChannelSubscription) => {
+    handleConfirmDelete = () => {
         this.setState({showConfirmModal: false});
-        this.deleteChannelSubscription(sub);
+        this.deleteChannelSubscription();
     }
 
     handleDeleteChannelSubscription = (sub: ChannelSubscription): void => {
-        this.setState({showConfirmModal: true});
-        this.setState({subscriptionToDelete: sub});
+        this.setState({
+            showConfirmModal: true,
+            subscriptionToDelete: sub,
+        });
     };
 
-    deleteChannelSubscription = (sub: ChannelSubscription): void => {
-        this.props.deleteChannelSubscription(sub).then((res: { error?: { message: string } }) => {
+    deleteChannelSubscription = (): void => {
```

Comment:
Sorry for the back and forth on this, but I actually think this one should take the subscription as a parameter. The main reason being is that it is basically a helper function here, so it makes sense to make it a pure function.

Reply:
For my clarification -- I thought it was weird accessing `this.state` inside the function, but technically didn't understand why.  Your explanation makes sense.  Thanks, @mickmister!